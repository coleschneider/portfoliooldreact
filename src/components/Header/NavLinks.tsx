import * as React from 'react'
import { useRouteMatch, Link } from 'react-router-dom'
import styled, { css } from 'styled-components'
import ResumePDF from '../../assets/Resume.pdf'

interface NavLinkProps {
  path: ConstantPaths
  display: string
  target?: string
}

const LinkStyles = css`
  z-index: 2;
  margin: 0 30px;
  cursor: pointer;
  line-height: 2.5em;
  text-decoration: none;
  color: #0f3351;
  ::before {
    content: '';
    position: relative;
    display: block;
    width: 100%;
    height: 2px;
    top: 18px;
    z-index: 7;
    left: 0;
    background-color: #0f3351;
    transform: scaleX(0);
    transition: all 0.65s cubic-bezier(0.7, 0.3, 0, 1);
    transform-origin: left;
    will-change: transform;
  }
  &:hover {
    &::after {
      transform: scale(1);
    }
  }
`
const TextLink = styled(({ matches, ...restProps }) => <Link {...restProps} />)`
  ${LinkStyles}
  ::after {
    content: '';
    position: relative;
    display: block;
    width: 100%;
    height: 2px;
    z-index: 8;
    left: 0;
    background-color: #0f3351;
    transform: ${({ matches }) => (matches ? 'scale(1)' : 'scale(0)')};
    transition: all 0.65s cubic-bezier(0.7, 0.3, 0, 1);
    transform-origin: right;
  }
`

const navPaths: NavLinkProps[] = [
  { path: '/', display: 'HOME' },
  { path: '/mywork', display: 'WORK' },
  { path: '/static/media/Resume.pdf', display: 'RESUME', target: '_blank' },
]

function NavLink({ display, path, ...rest }: NavLinkProps) {
  const isMatch = useRouteMatch(path)
  const isResume = path === '/static/media/Resume.pdf' && display === 'RESUME'
  return (
    <TextLink
      {...rest}
      to={path}
      matches={isMatch && isMatch.isExact}
      as={isResume ? 'a' : undefined}
      href={isResume || undefined}
    >
      {display}
    </TextLink>
  )
}
NavLink.displayName = 'NavLink'

function NavLinks() {
  return (
    <React.Fragment>
      {navPaths.map(props => (
        <NavLink key={props.display} {...props} />
      ))}
    </React.Fragment>
  )
}

NavLinks.displayName = 'NavLinks'
export default NavLinks
