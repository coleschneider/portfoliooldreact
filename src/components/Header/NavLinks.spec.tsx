import React from 'react'
import NavLinks from './NavLinks'

describe('NavLinks', () => {
  it('should match the snapshot', () => {
    const Wrapper = shallow(<NavLinks />)
    expect(Wrapper).toMatchSnapshot()
  })
  it('should include the navlinks at different media queries', () => {
    const Links = shallow(<NavLinks />).find('NavLink')
    expect(Links).toHaveLength(3)
    expect(Links.first().props()).toEqual({ path: '/', display: 'Home' })
    expect(Links.at(1).props()).toEqual({ path: '/work', display: 'Work' })
    expect(Links.at(2).props()).toEqual({ path: '/static/media/Resume.pdf', display: 'Resume', target: '_blank' })
  })
})
