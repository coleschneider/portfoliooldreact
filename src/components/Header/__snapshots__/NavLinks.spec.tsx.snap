// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavLinks should match the snapshot 1`] = `
<Fragment>
  <NavLink
    display="Home"
    key="Home"
    path="/"
  />
  <NavLink
    display="Work"
    key="Work"
    path="/work"
  />
  <NavLink
    display="Resume"
    key="Resume"
    path="/static/media/Resume.pdf"
    target="_blank"
  />
</Fragment>
`;
